clients/src/main/java/org/apache/kafka/common/KafkaFuture.java

```diff
@@ -31,7 +31,7 @@
      * A function which takes objects of type A and returns objects of type B.
      */
     public static abstract class Function<A, B> {
-        public abstract B apply(A a);
+        public abstract B apply(A a) throws Throwable;
```

Comment:
Not sure about this. Note that Java 8's Function doesn't specify any exceptions.